src/types.rs

```diff
@@ -638,6 +700,21 @@ impl HeapCellValue {
 
 const_assert!(mem::size_of::<HeapCellValue>() == 8);
 
+#[cfg(target_pointer_width="32")]
+#[bitfield]
+#[repr(u32)]
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
+pub struct UntypedArenaPtr {
+    ptr: B29,
+    #[allow(unused)] f: B1,
```

Comment:
This is  a departure from the 64-bit code because I wanted to try whether it's better to match this to ConsPtr, but apparently it doesn't affect anything. No more/less tests passed.